RecoLocalTracker/SiStripClusterizer/plugins/alpaka/SiStripRawToCluster.cc

```diff
@@ -0,0 +1,632 @@
+        //@pietroGru calculate the correct pointer inside the fedRawDataGPU where the detector's channel begins
+        auto my_offset =
+            fedBufferBlocksRaw_.getOffset(fedID)  // this is just an offset - same in host/dev mem
+            +
+            (fedChannel.data() -
+             raw_[fedi]
+                 ->data());  // this is the offset between the pointer where the channel data begins and the position of the data as a whole start (this difference is the same in host/device, and this is calculated on host - since both things are in host memory already)
+
+        rawPointerAddresses_onHost[i] =
+            // fedBufferBlocksRaw_.getData().data()       // this pointer is in the host memory (used in debug only)
+            fedBufferBlocksRaw_onDevice.data()  // this pointer is in the device memory
+            + my_offset;                        // this is the offset I need to add to reach the FEDChannel object
+
+        offset += (len - headerlen);
+        n_strips +=
+            (len -
+             headerlen);  //((len >= headerlen) ? len-headerlen : len); unsure if some memory here could be saved (need to check with sistrip conveners)
+
+#if defined(EDM_ML_DEBUG) && defined(SUPERDETAILS)
+        if (i % 100 == 0)
+          LogDebug("SiStripPrtArit") << i << "\t" << fedID << "\t" << (int)fedCH << fedi << "\t" << len << "\t" << off
+                                     << "\t" << my_offset << "\t" << offset << "\n";
+#endif
+      } else {
+        chanlocs_onHost.view()[i] = {nullptr, 0, 0, 0, READOUT_MODE_INVALID, 0, invalidFed, 0, invalidDet};
+        rawPointerAddresses_onHost[i] = nullptr;
+      }
+    }
+
+#ifdef EDM_ML_DEBUG
+    alpaka::wait(iEvent.queue());
+#endif
+
+    //
+    //
+    // @brief    Map Detector Channels to FED Data - END
+    /////////////////////////////////////////////////////////////////////////////
+
+    /////////////////////////////////////////////////////////////////////////////
+    // @brief    Run the algorithm for unpacking and clustering
+    //
+    //
+    // const int n_strips = offset;
+
+    // Make the cabling and clusterizer conditions available on device
+    auto clusterizerConditions_onDevice = SiStripClusterizerConditionsDevice(cablingMapData.sizes(), iEvent.queue());
+    alpaka::memcpy(iEvent.queue(), clusterizerConditions_onDevice.buffer(), cablingMapData.const_buffer());
+    // std::cout <<
+    //   "DetToFedsSoA size= " << cablingMapData.sizes()[0] << "\n"
+    //   "Data_fedchSoA size= " << cablingMapData.sizes()[1] << "\n"
+    //   "Data_stripSoA size= " << cablingMapData.sizes()[2] << "\n"
+    //   "Data_apvSoA size= " << cablingMapData.sizes()[3] << "\n";
+    // Make the mapping between the unpacked FED data and strips available on the device
+    auto chanlocs_onDevice = SiStripMappingDevice(chanlocs_onHost->metadata().size(), iEvent.queue());
+    alpaka::memcpy(iEvent.queue(), chanlocs_onDevice.buffer(), chanlocs_onHost.const_buffer());
```

Comment:
If the `print_SiStripDataCompare_()` could be moved before this `memcpy()`, these two lines could be replaced with
```suggestion
    auto chanlocs_onDevice = cms::alpakatools::moveToDeviceAsync(iEvent.queue(), std::move(chanlocs_onHost));
```
(for more information see https://github.com/cms-sw/cmssw/blob/master/HeterogeneousCore/AlpakaInterface/README.md#host-to-device-move)

The `moveToDeviceAsync()` would bring two benefits
* It would be clear that `chanlocs_onHost` should not be modified after that line (accesses to its internal memory would likely lead to segfaults)
* The `memcpy()` would be omitted for the cpu backend (i.e. on cpu backend the contents of `chanlocs_onHost` would be moved as-is to `chanlocs_onDevice`)

Reply:
`print_SiStripDataCompare_()`  was supposed to compare the results of the device unpacking (after `algo_.unpackStrips`) with an unpacking made on the host. It is a residual of early development stages. I will remove it and think about a more clever way to have debugging of the unpacking process.